### PR TITLE
Feature: Auto Publish Client SDK to npm

### PR DESCRIPTION
**Pull request by MartinMinkov**
This PR adds the functionality to publish our client sdk to npm. The following was added:

1. Created a `ClientSdk.dhall` file to specify the job to run when we want to release to npm
2. Update the `client_sdk-tool.sh` to write the npm token to `~/.npmrc`
3. Update terraform to specify the npm token as an aws secret
4. Update terraform to specify the npm token as an environment variable

EDIT:
Seems to be running okay in our CI jobs.
image

### CODE REVIEW

**Review by 0x0I — APPROVED** [review]

**Comment by aneesharaines** [comment]
!approved-for-mainnet

**Comment by aneesharaines** [comment]
!approved-for-mainnet

File: buildkite/src/Jobs/Release/ClientSdk.dhall [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+in
+
+Pipeline.build
+  Pipeline.Config::{
+    spec = 
+      JobSpec::{
+        dirtyWhen = OpamInit.dirtyWhen # [
+          S.strictlyStart (S.contains "buildkite/src/Jobs/Release/ClientSdk"),
+          S.strictlyStart (S.contains "src")
```
Comment (bkase) [inline buildkite/src/Jobs/Release/ClientSdk.dhall:22]: can you add another line for `S.strictlyStart (S.contains "frontend/client_sdk")`

**Review by bkase — APPROVED** [review]

File: buildkite/src/Jobs/Release/ClientSdk.dhall (hunk 1 above)
Comment (MartinMinkov) [reply buildkite/src/Jobs/Release/ClientSdk.dhall:23]: Done.

**Pull request merged**